golang/trace/zipin_tracereporter_test.go

```diff
@@ -126,9 +126,21 @@ func TestSubmit(t *testing.T) {
 		thriftSpan.BinaryAnnotations = []*gen.BinaryAnnotation{}
 		ret := &gen.Response{Ok: true}
 
-		args.s.On("Submit", ctxArg(), thriftSpan).Return(ret, nil)
-		err := args.c.Report(span, annotations, nil, endpoint)
-		assert.NoError(t, err)
+		called := make(chan struct{})
+		args.s.On("Submit", ctxArg(), thriftSpan).Return(ret, nil).Run(func(arg mock.Arguments) {
+			close(called)
+		})
+		got, err := args.s.Submit(ctx, thriftSpan)
```

Comment:
why are you calling Submit on the server? You should only be testing `args.c.Report`